ui/app/models/kv/config.js

```diff
@@ -0,0 +1,31 @@
+import Model, { attr } from '@ember-data/model';
+import lazyCapabilities, { apiPath } from 'vault/macros/lazy-capabilities';
+import { withFormFields } from 'vault/decorators/model-form-fields';
+
+// This model is used only for display only - configuration happens via secret-engine model when an engine is mounted
+@withFormFields()
+export default class KvConfigModel extends Model {
+  @attr('number', { label: 'Maximum number of versions' }) maxVersions;
+
+  @attr('boolean', { label: 'Require check and set' }) casRequired;
+
+  @attr({ label: 'Automate secret deletion' }) deleteVersionAfter;
+
+  @lazyCapabilities(apiPath`${'backend'}/config`, 'backend') configPath;
+
+  get canRead() {
+    return this.configPath.get('canRead') !== false;
+  }
+
+  // used in template to render using this model instead of secret-engine (where these attrs also exist)
+  get displayFields() {
+    return ['casRequired', 'deleteVersionAfter', 'maxVersions'];
+  }
+
+  get isNeverDelete() {
```

Comment:
This may feel a bit wonky to do here but it didn't feel necessary to make a component class for this minor attr display when I imagine this will be refactored when KV gets a redesign and we follow the LDAP pattern [PR](https://github.com/hashicorp/vault/pull/21430) of toggling the mount configs

Reply:
Could we instead use this as a display value, and then in line 29 instead of false we would return formatted `this.deleteVersionAfter`? It would clean up the template logic a bit

Reply:
I originally tried that but the format-duration helper stopped me 😓